### ISSUE TITLE
Why structs instead classes?

### ISSUE THREAD

**Issue by turric4n** [megamarc/CsTilengine#2]
Hi Marc,

I'm reviewing the code and all entities are defined using struct instead classes, there is a specific reason to do this? 

Also, I found this library binding : 

`[DllImport("Tilengine")]
private static extern IntPtr TLN_CreateTilemap(int rows, int cols, Tile[] tiles, uint bgcolor, Tileset tileset);` 

You pass Tileset structure reference directly instead handle pointer. I'm not sure if this is ok, because I didn't test it. Can you confirm?

Thanks

**Comment by megamarc** [comment]
Hi!
Structures are lightweight value types, in most cases they just hold a single integer inside the binding. Classes have the overhead of full memory management, garbage collection ,etc. Being just containers for the underlying memory handlers of the library, I choose structs.
I have to check your observation, of course there may be a mistake, thanks for pointing it!

**Comment by turric4n** [comment]
Hi Marc,
Thanks for your response.

**Comment by Shadowblitz16** [comment]
You can use idisposable and finalizers if you use classes so users don't have to worry about memory management.

**Comment by megamarc** [comment]
Yes and no. As CsTilengine is just a thin binding, special care must be taken about reference ownership. For example, a reference obtained with TLN_LoadPalette() can be automatically disposed when the object container is destroyed. But a reference obtained with TLN_GetBitmapPalete() (for example) must not be disposed when the object container is destroyed, as the palette is owned by the bitmap. So ownership management should be implemented. Also keep in mind that the overhead of classes at runtime is greater than structs.

**Comment by turric4n** [comment]
Structs are fine in this scenario, no benefits of classes here.
